index.rst

```diff
@@ -177,43 +193,44 @@ This class is a subclass of ``dmlc::Parameter``.
 .. parsed-literal::
 
   {
-    "num_roots" : *string*,
-    "num_nodes" : *string*,
-    "num_deleted" : *string*,
-    "max_depth" : *string*,
-    "num_feature" : *string*,
-    "size_leaf_vector" : *string*
+    "num_deleted" : *integer*,
+    "max_depth" : *integer*,
   }
 
 Node
 ----
+Each node is represented as a JSON array of a fixed size, each element being
+determined by the ``node_schema`` field of XGBoostModel_.
+
 .. parsed-literal::
 
-  {
-    "parent" : *integer*,
-    "cleft" : *integer*,
-    "cright" : *integer*,
-    "sindex" : *unsigned integer*,
-    "float_info" : *floating-point*
-  }
+  [
+    *integer* (parent_id),
+    *integer* (child_left_id),
+    *integer* (child_right_id),
+    *unsigned integer* (split_index),
+    *boolean* (default_left),
+    *floating-point* (float_info)
```

Comment:
Could you clarify what it is?

Reply:
It is described in the full document. It doubles as both leaf output and split threshold.

Reply:
A leaf doesn't need `child_*_id` ,`split_index` and `default_left`. So I propose to keep leaves not as objects, but as simple floats and rename this field to `margin`.

Reply:
>How do we tell leaf nodes apart from non leaf nodes, now that we got rid of the field names for the sake of compactness?

By type or length, I guess.

```javascript
[
  [0, 1, 2, 0, 1, 0.5],
  [0, 3, 4, 1, 1, 0.5],
  [0, 5, 6, 1, 1, 0.5],
  [1, 7.], [1,8.], [2,9.], [2, 0.]
]
```

if we keep parent indices,

[
  [1, 2, 0, 1, 0.5],
  [3, 4, 1, 1, 0.5],
  [5, 6, 1, 1, 0.5],
  7., 8., 9., 0.1
]
```

if we dont

```
[
 [7., 8., 1, 1, 0.5],
 [9., 0.1, 1, 1, 0.5],
 0, 1, 0.5
]
```

if we encode into structure

Reply:
I’ll update the PR to distinguish leaf nodes from non leaf nodes.